src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Helpers.cs

```diff
@@ -37,12 +37,6 @@ internal static bool OriginalSymbolsMatch(
         if (searchSymbol == null || symbolToMatch == null)
             return false;
 
-        // Avoid the expensive checks if we can fast path when the compiler just says these are equal. Also, for the
-        // purposes of symbol finding nullability of symbols doesn't affect things, so just use the default
-        // comparison.
-        if (searchSymbol.Equals(symbolToMatch))
```

Comment:
This is not related to the core fix, but when I traced down the problem I noticed that `OriginalSymbolsMatchCore` method, which is invoked right below this check, has exactly the same condition in it, so there is clearly duplication of efforts going on here)